CONTRIBUTING.md

```diff
@@ -1,49 +1,151 @@
+
+Some technical terms can be intuitively translated into French. For instance, we will translate "tag" to "balise" or "generator" to "générateur" and that won't be chocking for the reader. We should translate as soon as it's intuitive enough. 
+
+On the other hand, some terms are not intuitive at all to translate and we generally use the English one. For instance, we wouldn't naturally say "assistant" for "helper" or "fabrique" for "factory".
+
+In any case, when you encounter a technical term:
+- Put the English in _italic_ style and keep the regular style for French.
+- The first time the word appears on the page, use the language you'll reuse along the page and add the language you _won't_ use in parenthesis so the reader can do the link between English and French.
+
+In this example, we will stick to "espace de nom" in the rest of the page:
+```diff
+- Components like these are known as namespaced components. Namespacing allows us to organize our components by folders according to their purpose.
++ Les composants comme ceux-ci sont appelés composants à espace de nom (_namespaced components_). Les espaces de nom permettent d'organiser les composants par dossiers selon leur fonction.
 ```
-git remote add upstream https://github.com/dazzlingfugu/ember-fr-guides-source.git
-git checkout master
-git pull upstream master
+
+In this example, we will go for "store" and "record" in the rest of the page:
+```diff
+- The EmberData store provides an interface for retrieving records of a single type. Use `store.findRecord()` to retrieve a record.
++ Le _store_ (magasin, réserve) d'EmberData fournit une interface pour retrouver les _records_ (enregistrements) d'un seul type. Utilisez `store.findRecord()` pour récupérer un _record_.
+```
+
+If the literal translation of a term is _reeeeally_ convoluted, don't hesitate to be wordy about it. The purpose is to make this understandable to non-English speakers:
+
+```diff
+- This method is also known as the _model hook_.
++ Cette méthode est aussi appelée le _model hook_ (_hook_ signifie littéralement "un crochet" mais se traduit mal dans un contexte technique. On peut voir le _hook_ comme une fonction mise à disposition et exécutée en interne par le framework, et dont on définit le contenu, on y "accroche" du code).
+```
+
+### 3. Don't translate articles titles / translate descriptions
+
+Some pages of the Guides reference articles, introduced with a title and sometimes a description. If this resource is also available in French (for instance, a lot of pages from MDN website are translated), then use the French URL in the translation (be careful with that MDN example though, because some pages are not translated entirely and it can be better to keep the English).
+
+If the resource is not translated or if the translation is incomplete, keep the English link and translate only the description, not the title of the resource itself, because we need to make clear to the reader that it's an English document:
```

Comment:
```suggestion
If the resource is not translated or if the translation is incomplete, keep the English link and translate only the description, not the title of the resource itself, because we need to make clear to the reader it's an English document:
```